### PR TITLE
Artifacts now react to water. Huzzah

### PR DESCRIPTION
**Pull request by quardbreak**
credits to CrimsonShrike

### CODE REVIEW

**Comment by CrimsonShrike** [comment]
Doesnt nebula have a chem fluids? Fairly sure some changed will be involved

File: code/modules/xenoarcheaology/artifacts/artifact.dm [hunk 1]
```diff
@@ -3,8 +3,9 @@
 	desc = "A large alien device."
 	icon = 'icons/obj/xenoarchaeology.dmi'
 	icon_state = "ano00"
+	density = TRUE
```
Comment (CrimsonShrike) [inline code/modules/xenoarcheaology/artifacts/artifact.dm:6]: Change seems unrelated

File: code/modules/xenoarcheaology/artifacts/artifact.dm [hunk 2]
```diff
@@ -129,3 +130,6 @@
 		out += secondary_effect.getDescription()
 
 	return out
+
+/obj/structure/artifact/fluid_act(depth)
```
Comment (CrimsonShrike) [inline code/modules/xenoarcheaology/artifacts/artifact.dm]:
```suggestion
/obj/structure/artifact/fluid_act(var/datum/reagents/fluids)
    check_triggers(/datum/artifact_trigger/proc/on_fluid_act, fluids)
```

File: code/modules/xenoarcheaology/artifacts/triggers/_trigger.dm [hunk 3]
```diff
@@ -19,6 +19,9 @@
 /datum/artifact_trigger/proc/on_bump(atom/movable/AM)
 	return FALSE
 
+/datum/artifact_trigger/proc/on_water_act(depth)
```
Comment (CrimsonShrike) [inline code/modules/xenoarcheaology/artifacts/triggers/_trigger.dm]:
```suggestion
/datum/artifact_trigger/proc/on_fluid_act(var/datum/reagents/fluids)
```

File: code/modules/xenoarcheaology/artifacts/artifact.dm (hunk 1 above)
Comment (CrimsonShrike) [inline code/modules/xenoarcheaology/artifacts/artifact.dm:6]: Make sure the waterproof var is unset. Although I assume it may be.

File: code/modules/xenoarcheaology/artifacts/triggers/chemical.dm [hunk 4]
```diff
@@ -24,6 +24,10 @@
 		/decl/material/solid/ice
 	)
 
+/datum/artifact_trigger/chemical/water/on_water_act(depth)
```
Comment (CrimsonShrike) [inline code/modules/xenoarcheaology/artifacts/triggers/chemical.dm]:
```suggestion
/datum/artifact_trigger/chemical/on_fluid_act(var/datum/reagents/fluids)
    for(var/reagent in required_chemicals)
			if(fluids.remove_reagent(reagent, 5))
				return TRUE
	return ..()

```

File: code/modules/xenoarcheaology/artifacts/triggers/chemical.dm [hunk 5]
```diff
@@ -24,6 +24,10 @@
...
+	if(depth > FLUID_EVAPORATION_POINT)
```
Comment (CrimsonShrike) [inline code/modules/xenoarcheaology/artifacts/triggers/chemical.dm]: Not sure how ot make a suggestion span multiple lines, so just remove this part

**Review by CrimsonShrike — CHANGES_REQUESTED** [review]

File: code/modules/xenoarcheaology/artifacts/artifact.dm (hunk 1 above)
Comment (quardbreak) [reply code/modules/xenoarcheaology/artifacts/artifact.dm:6]: structures don't have waterproof var

**Comment by MistakeNot4892** [comment]
Crimson's right, this code won't work as written.

**Review by MistakeNot4892 — APPROVED** [review]
